From linux-usb-owner@vger.kernel.org  Tue Oct  5 13:30:26 2010
From: Felipe Balbi <balbi@ti.com>
To: Greg KH <greg@kroah.com>
Cc: Linux USB Mailing List <linux-usb@vger.kernel.org>,
	Sergei Shtylyov <sshtylyov@ru.mvista.com>,
	Felipe Balbi <balbi@ti.com>
Subject: usb: musb: gadget: kill duplicate code in musb_gadget_queue()
Date: Fri, 24 Sep 2010 13:44:11 +0300
Message-Id: <1285325055-1247-11-git-send-email-balbi@ti.com>

From: Sergei Shtylyov <sshtylyov@ru.mvista.com>

I've noticed that musb_gadget_queue() checks for '!req->buf' condition twice:
in the second case the code is both duplicate and unreachable as the first
check returns early.

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/musb/musb_gadget.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/usb/musb/musb_gadget.c
+++ b/drivers/usb/musb/musb_gadget.c
@@ -1214,8 +1214,6 @@ static int musb_gadget_queue(struct usb_
 
 	if (is_dma_capable() && musb_ep->dma)
 		map_dma_buffer(request, musb);
-	else if (!req->buf)
-		return -ENODATA;
 	else
 		request->mapped = 0;
 
